class.jetpack.php

```diff
@@ -7067,4 +7068,29 @@ function opt_in_jetpack_manage_notice() {
 	function can_display_jetpack_manage_notice() {
 		_deprecated_function( __METHOD__, 'jetpack-7.3' );
 	}
+
+	/**
+	 * Clean leftoveruser meta.
+	 *
+	 * Delete Jetpack-related user meta when it is no longer needed.
+	 *
+	 * @since 7.3.0
+	 *
+	 * @param int $user_id User ID being updated.
+	 */
+	public static function user_meta_cleanup( $user_id ) {
+		$meta_keys = array(
+			// AtD removed from Jetpack 7.3
+			'AtD_options',
+			'AtD_check_when',
+			'AtD_guess_lang',
+			'AtD_ignored_phrases',
+		);
+
+		foreach ( $meta_keys as $meta_key ) {
+			if ( get_user_meta( $user_id, $meta_key ) ) {
+				delete_user_meta( $user_id, $meta_key );
```

Comment:
Maybe we can just delete the user meta data without checking if it exists. We would avoid a database transaction and in any case, it's the same for the database to get or to delete.

Reply:
Considering that all the core functions have some overhead, I'll be more concerned about how many times the `profile_update` hook gets called.

Looking at the internals, `get_user_meta` uses `get_metadata` that uses `wp_cache_get`, but `delete_user_meta` internally uses `delete_metadata` that performs a few checks on the db before `DELETE`. So, `get_user_meta` not always queries the DB.

Reply:
👍

Reply:
Yeah, @rodrigoi captures my thinking -- we can sometimes save from the db call when it can check in the cache if it is present or not.

The hook is called on `wp_insert_user` when updating an existing user (does not when adding a new user), so any time the user itself is updated (password change, editing the profile, etc).